rustual-boy-middleware/src/color.rs

```diff
@@ -0,0 +1,69 @@
+//! Some color utilities that are useful for implementing the anaglyph modes
+
+/// Represents a color
+pub struct Color {
+    r: f32,
```

Comment:
I think `u8` for the internal storage here would've been more appropriate. I can understand the argument that perhaps `f32`'s might be more clear when expressing individual colors, but `u8`'s are going to use less storage and allow us to optimize in ways I don't think we can with `f32`'s (and we can still have `From<(f32, f32, f32)>` like you have below with reasonable precision).